homeassistant/components/hive/config_flow.py

```diff
@@ -90,24 +92,48 @@ async def async_step_2fa(self, user_input=None):
 
             if not errors:
                 try:
+                    if self.context["source"] == config_entries.SOURCE_REAUTH:
+                        return await self.async_setup_hive_entry()
                     self.device_registration = True
-                    return await self.async_setup_hive_entry()
+                    return await self.async_step_configuration()
                 except UnknownHiveError:
                     errors["base"] = "unknown"
 
         schema = vol.Schema({vol.Required(CONF_CODE): str})
         return self.async_show_form(step_id="2fa", data_schema=schema, errors=errors)
 
+    async def async_step_configuration(self, user_input=None):
+        """Handle hive configuration step."""
+        errors = {}
+
+        if user_input:
+            if not errors:
```

Comment:
I think this condition is not needed because you already assigned errors to an empty dict above.

Reply:
Done